### PR TITLE
Implement configurable retries

### PR DESCRIPTION
**Pull request by MarcoBgn**

### CODE REVIEW

File: lib/fieldhand/paginator.rb [hunk 1]
```diff
@@ -11,20 +11,25 @@ module Fieldhand
   #
   # See https://www.openarchives.org/OAI/openarchivesprotocol.html#FlowControl
   class Paginator
-    attr_reader :uri, :logger, :timeout, :headers, :http
+    attr_reader :uri, :logger, :timeout, :interval, :headers, :http
+    attr_accessor :retries
```
Comment (jbilbo) [inline lib/fieldhand/paginator.rb]: any reason why retries is specifically an accessor (= you want write access outside of the initialize)?

File: spec/fieldhand/paginator_spec.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ module Fieldhand
       it 'raises a Bad Argument Error if returned from the repository' do
         stub_oai_request('http://www.example.com/oai?verb=Identify&bad=Argument',
                          'bad_argument_error.xml')
-        paginator = described_class.new('http://www.example.com/oai')
+        paginator = described_class.new('http://www.example.com/oai', retries: 0)
```
Comment (jbilbo) [inline spec/fieldhand/paginator_spec.rb]: shall we make `0` as default value for retries to be fully backward compatible. So no defining anything leaves the code as it was before this change.

Comment (MarcoBgn) [reply spec/fieldhand/paginator_spec.rb]: This is a very good point, yes 👍

File: lib/fieldhand/paginator.rb (hunk 1 above)
Comment (MarcoBgn) [reply lib/fieldhand/paginator.rb]: It's in order to decrease the value in `retries -= 1`

**Pull request merged**